Author: Pino Toscano <pino@debian.org>
Description: Make QtWebEngine an optional requirement
 This allows to build konqueror, albeith without an HTML component, also on
 architectures that do not have QtWebEngine.
Forwarded: no
Last-Update: 2017-12-15

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -23,7 +23,8 @@ include(ECMQtDeclareLoggingCategory)
 set(KONQUEROR_LIB_VERSION "5.0.97")
 set(KONQUEROR_VERSION "${KONQUEROR_LIB_VERSION}")
 
-find_package(Qt5 ${QT_MIN_VERSION} REQUIRED COMPONENTS Core Widgets WebEngineWidgets)
+find_package(Qt5 ${QT_MIN_VERSION} REQUIRED COMPONENTS Core Widgets)
+find_package(Qt5 ${QT_MIN_VERSION} COMPONENTS WebEngineWidgets)
 find_package(KF5 ${KF5_MIN_VERSION} REQUIRED COMPONENTS Parts KCMUtils KHtml KDELibs4Support Archive Crash)
 
 find_package(KF5 ${KF5_MIN_VERSION} COMPONENTS Activities DocTools) # Optional
@@ -53,7 +54,9 @@ add_subdirectory( libkonq )
 add_subdirectory( src )
 add_subdirectory( client )
 add_subdirectory( autotests )
-add_subdirectory( webenginepart )
+if (Qt5WebEngineWidgets_FOUND)
+    add_subdirectory( webenginepart )
+endif()
 
 add_subdirectory( about )
 add_subdirectory( pics )
--- a/autotests/CMakeLists.txt
+++ b/autotests/CMakeLists.txt
@@ -7,12 +7,14 @@ include_directories( ${CMAKE_CURRENT_SOU
 
 ########### konqviewmgrtest ###############
 
-add_executable(konqviewmgrtest konqviewmgrtest.cpp)
-add_test(konqviewmgrtest konqviewmgrtest)
-ecm_mark_as_test(konqviewmgrtest)
-target_link_libraries(konqviewmgrtest kdeinit_konqueror Qt5::Core Qt5::Gui
-   # KF5::KHtml
-   kwebenginepartlib Qt5::WebEngineWidgets Qt5::Test)
+if (Qt5WebEngineWidgets_FOUND)
+  add_executable(konqviewmgrtest konqviewmgrtest.cpp)
+  add_test(konqviewmgrtest konqviewmgrtest)
+  ecm_mark_as_test(konqviewmgrtest)
+  target_link_libraries(konqviewmgrtest kdeinit_konqueror Qt5::Core Qt5::Gui
+     # KF5::KHtml
+     kwebenginepartlib Qt5::WebEngineWidgets Qt5::Test)
+endif()
 
 ########### historymanagertest ###############
 
@@ -30,10 +32,12 @@ target_link_libraries(undomanagertest kd
 
 ########### konqhtmltest ###############
 
-add_executable(konqhtmltest konqhtmltest.cpp)
-add_test(konqhtmltest konqhtmltest)
-ecm_mark_as_test(konqhtmltest)
-target_link_libraries(konqhtmltest kdeinit_konqueror kwebenginepartlib Qt5::Core Qt5::Test)
+if (Qt5WebEngineWidgets_FOUND)
+  add_executable(konqhtmltest konqhtmltest.cpp)
+  add_test(konqhtmltest konqhtmltest)
+  ecm_mark_as_test(konqhtmltest)
+  target_link_libraries(konqhtmltest kdeinit_konqueror kwebenginepartlib Qt5::Core Qt5::Test)
+endif()
 
 ########### konqviewtest ###############
 
